kernel/src/process.rs

```diff
@@ -330,15 +330,56 @@ impl<'a> Process<'a> {
         if data_len.count_ones() != 1 {
             panic!("Tock MPU does not currently handle complex region sizes");
         }
-        let data_region_len = math::log_base_two(data_len as u32);
+        let data_region_len = math::PowerOfTwo::from(data_len as u32);
 
+        // Text segment read/execute (no write)
         let text_start = self.text.as_ptr() as usize;
         let text_len = self.text.len();
         if text_len.count_ones() != 1 {
             panic!("Tock MPU does not currently handle complex region sizes");
         }
-        let text_region_len = math::log_base_two(text_len as u32);
 
+        // Invariant: text_len is a power of two
+        if text_start % text_len != 0 {
+            // Text length not aligned to text start
+            let div = text_start % text_len;
+            let region_size = div * 8; // there are 8 subregions in a region
```

Comment:
We should enforce that apps are ordered by size somewhere, otherwise a small app followed by a large app will break (i.e.: 
`text_start  = 0x30100`
`text_len    = 0x10000`
) as the region size will be < the text size